Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -183,6 +183,11 @@ def bump_formula_pr
     elsif !new_url && !new_version
       odie "#{formula}: no --url= or --version= argument specified!"
     else
+      new_url ||= if old_url.start_with?(PyPI::PYTHONHOSTED_URL_PREFIX)
+        package_name = File.basename(old_url).match(/^(.+)-[a-z\d.]+$/)[1]
+        _, url = PyPI.get_pypi_info(package_name, new_version)
+        url
+      end
```

Comment:
This logic could arguably be in a helper function here or on `PyPI` which returns `nil` if it doesn't match?

Reply:
I'm not sure which part you're talking about. `File.basename(old_url).match(/^(.+)-[a-z\d.]+$/)[1]`?

I could add a method to `PyPI` that takes in a url + version and outputs the new url or `nil`.

Then, this would become something like:

```ruby
new_url ||= PyPI.update_pypi_url(old_url, new_version)
```

Reply:
> Then, this would become something like:

Yup, that's what I was thinking!